src/main/java/bio/terra/service/filedata/google/gcs/GcsPdao.java

```diff
@@ -327,22 +334,56 @@ private void fileAclOp(AclOp op, Dataset dataset, List<String> fileIds, Map<IamR
     }
 
     /**
-     * Extract the bucket from a gs path.
+     * Extract the path portion (everything after the bucket name and it's trailing slash) of a gs path.
      */
-    public static String extractBucketFromPath(final String path) {
-        if (!path.startsWith("gs://") && StringUtils.countMatches(path, ',') < 3) {
-            throw new PdaoException(String.format("Bad GS path %s", path));
-        }
-        // Given a string that looks like "gs://some bucket/pat1/path2" (format verified above), this will extract
-        // "some bucket" and return
-        return path.substring(5).split("/")[0];
+    private static String extractFilePathInBucket(final String path, final String bucketName) {
+        return StringUtils.removeStart(path, String.format("gs://%s/", bucketName));
```

Comment:
Is this different than `getGcsLocatorFromGsPath(gspath).getPath()`?

Reply:
yeah, I like the fact that this takes in an explicit bucket.  I was debating migrating but liked that this was using a known bucket name